### DISCUSSION TITLE
How to calculate the scattered potential by a body around it and at a distance |r| from it ?

### DISCUSSION THREAD

**Discussion by KapilKhanal** [capytaine/capytaine#363] [General]
I am assuming diffraction potential is the scattered potential around the body but does it decay as the distance from the body increases? There is 'get_potential_on_mesh' method on the solver object and that takes in the xyz to calculate the potential. Does this mean that we can provide any xyz (not necessarily on the mesh) to calculate the scattered potential at that location. I am trying to calculate the potential at one body due to another body not too far away (and without joining the body).

**Comment by mancellin** [comment]
> Does this mean that we can provide any xyz (not necessarily on the mesh) to calculate the scattered potential at that location

Yes. The latest version (released last week) actually includes a new method `solver.compute_potential` that does basically the same thing as `get_potential_on_mesh` but does not require a mesh. You can directly pass a list of points and get the value of the potential at these points. (See the post-processing section of the documentation.)

> I am assuming diffraction potential is the scattered potential around the body but does it decay as the distance from the body increases?

Yes, in Capytaine, the diffraction potential does not include the incoming waves potential and should decay with the distance. (Unlike in WAMIT, where I believe the term "diffraction" denotes the sum of both the "scattered" and the incoming waves potential.)

**Reply by KapilKhanal** [reply]
That makes sense. Thank you!

**Reply by mancellin** [reply]
PS: there is an unsolved inconsistency in the value of the potential between Capytaine and Nemoh (see #191). I would suggest you double check the results with another source if you can. And share them with us to help solve #191 :)
